relay-config/src/config.rs

```diff
@@ -277,6 +277,38 @@ impl ConfigObject for Credentials {
     }
 }
 
+/// Information on a downstream Relay.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct StaticRelayInfo {
```

Comment:
Ideally we manage to share the pre-existing `RelayInfo` type instead of creating a second type. That said, RelayInfo from the API does not contain the `id`, so let's stick with a separate type for now.

Reply:
Follow-up after reviewing https://github.com/getsentry/sentry/pull/26155: Since we use a mapping by Relay-ID there, I propose the following:

- Move `relay_server::actors::RelayInfo` here, and remove the `StaticRelayInfo` type.
- Change the config file to be `HashMap<RelayId, RelayInfo>` or `BTreeMap<RelayId, RelayInfo>`.
- Update `RelayCache` constructor to clone the mapping rather than recollect it.

Reply:
Ok, will do, I explicitly made it a Vector and convert it in RelayCache because I think it is a lot nicer to have a list in the configuration file than a dictionary (where it is not very explicit what the key represents).